test/tests/pbs_accumulate_resc_used.py

```diff
@@ -0,0 +1,1456 @@
+    e.job.resources_used["stra"] = '"glad,elated","happy"'
+    e.job.resources_used["foo_str3"] = \
+      \"\"\"{"a":6,"b":"some value #$%^&*@","c":54.4,"d":"32.5gb"}\"\"\"
+    e.job.resources_used["foo_str2"] = "seven"
+    e.job.resources_used["foo_str4"] = "eight"
+else:
+    e.job.resources_used["vmem"] = pbs.size("12gb")
+    e.job.resources_used["foo_i"] = 12
+    e.job.resources_used["foo_f"] = 0.12
+    e.job.resources_used["foo_str"] = '{"eight":8,"nine":9}'
+    e.job.resources_used["foo_str2"] = '{"seven":7}'
+    e.job.resources_used["cput"] = 12
+    e.job.resources_used["stra"] = '"cucumbers,bananas"'
+    e.job.resources_used["foo_str3"] = \"\"\""vn1":4,"vn2":5,"vn3":6\"\"\"
+"""
+
+        hook_name = "prolo"
+        a = {'event': "execjob_prologue", 'enabled': 'True'}
+        rv = self.server.create_import_hook(
+            hook_name,
+            a,
+            hook_body,
+            overwrite=True)
+        self.assertTrue(rv)
+
+        a = {'job_history_enable': 'True'}
+        self.server.manager(MGR_CMD_SET, SERVER, a, expect=True)
+
+        a = {'Resource_List.select': '3:ncpus=1', 'Resource_List.walltime': 10}
+        j = Job(TEST_USER)
+        j.set_attributes(a)
+        j.set_sleep_time("10")
+
+        # The pbsdsh call is what allows a first task to get spawned on
+        # on a sister mom, causing the execjob_prologue hook to execute.
+        j.create_script(
+            "pbsdsh -n 1 hostname\n" + "pbsdsh -n 2 hostname\n" + "sleep 10\n")
+
+        jid = self.server.submit(j)
+
+        # The results should show results for custom resources 'foo_i',
+        # 'foo_f', 'foo_str', 'foo_str3', and bultin resources 'vmem',
+        # 'cput', and should be accumulating  based
+        # on the hook script, where MS defines 1 value, while the 2 sister
+        # Moms define the same value. For 'string' type, it will be a
+        # union of all values obtained from sister moms and local mom, and
+        # the result will be in JSON-format.
+        #
+        # foo_str is for testing normal values.
+        # foo_str2 is for testing non-JSON format value received from MS.
+        # foo_str3 is for testing non-JSON format value received from a sister
+        # mom.
+        # foo_str4 is for testing MS-only set values.
+        #
+        # For string_array type  resource 'stra', it is not accumulated but
+        # will be set to last seen value from a mom prologue hook.
+        self.server.expect(JOB, {
+            'job_state': 'F',
+            'resources_used.foo_f': '0.35',
+            'resources_used.foo_i': '35',
+            'resources_used.foo_str4': "eight",
+            'resources_used.stra': "\"glad,elated\",\"happy\"",
+            'resources_used.vmem': '35gb',
+            'resources_used.cput': '00:00:35',
+            'resources_used.ncpus': '3'},
+            extend='x', offset=10, id=jid)
+
+        foo_str_dict_in = eval('{"eight": 8, "seven": 7, "nine": 9}')
```

Comment:
Previously you just set a similar variable to a dictionary.  Is there some reason to use eval() now?